spec/acceptance/ios_interface_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'spec_helper_acceptance'
+
+describe 'ios_interface' do
+  target = if ['3650', '3750', '2960'].include?(device_model)
+             'GigabitEthernet1/0'
+           elsif ['4507', '4948', '6503'].include?(device_model)
+             'GigabitEthernet1'
+           elsif ['4503'].include?(device_model)
+             'GigabitEthernet2'
+           else
+             'GigabitEthernet0'
+           end
+  it 'Set Two Instances' do
+    mac = if ['2960', '4507', '6503'].include?(device_model)
+            nil
+          else
+            ['mac_notification_added => true,', 'mac_notification_removed => true,',
+             'mac_notification_added => false,', 'mac_notification_removed => false,']
+          end
+    pp = <<-EOS
+    ios_interface { '#{target}/1':
+      #{mac[2]}
```

Comment:
This will fail on `2960, 4507 and 6503` as `mac` will be nil for these devices, maybe something like `#{mac ? mac[2] : ''}` will work to avoid this

or instead of having nil for the above devices just have it return something like `['', '', '', '']` so array indexing will not fall over and these params won't be used on these devices.

Reply:
updated